loadtest/control/simulcontroller/actions.go

```diff
@@ -1282,3 +1300,357 @@ func sendTypingEventIfEnabled(u user.User, channelId string) error {
+	threads, err := u.Store().ThreadsSorted(false, false)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if len(threads) == 0 {
+		threads, err = u.GetUserThreads(team.Id, &model.GetUserThreadsOpts{
+			PageSize:   25,
+			Extended:   false,
+			Deleted:    false,
+			Unread:     false,
+			Since:      0,
+			TotalsOnly: false,
+		})
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+	}
+
+	oldestThreadId := threads[len(threads)-1].PostId
+	// scrolling between 1 and 3 times
+	numScrolls := rand.Intn(3) + 1
+	for i := 0; i < numScrolls; i++ {
+		threads, err = u.GetUserThreads(team.Id, &model.GetUserThreadsOpts{
+			PageSize:   25,
+			Extended:   false,
+			Deleted:    false,
+			Unread:     false,
+			Since:      0,
+			TotalsOnly: false,
+			Before:     oldestThreadId,
+		})
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		oldestThreadId = threads[len(threads)-1].PostId
+		// idle time between scrolls, between 1 and 10 seconds.
+		idleTime := time.Duration(1+rand.Intn(10)) * time.Second
+		select {
+		case <-c.stopChan:
+			return control.UserActionResponse{Info: "action canceled"}
+		case <-time.After(idleTime):
+		}
+	}
+
+	// Switch to "Unread" tabs
+	unreadThreads, err := u.Store().ThreadsSorted(true, false)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if len(unreadThreads) == 0 {
+		unreadThreads, err = u.GetUserThreads(team.Id, &model.GetUserThreadsOpts{
+			PageSize:   25,
+			Extended:   false,
+			Deleted:    false,
+			Unread:     true,
+			Since:      0,
+			TotalsOnly: false,
+		})
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+	}
+
+	oldestUnreadThreadId := unreadThreads[len(unreadThreads)-1].PostId
+	// scrolling between 1 and 3 times
+	numScrolls = rand.Intn(3) + 1
+	for i := 0; i < numScrolls; i++ {
+		unreadThreads, err = u.GetUserThreads(team.Id, &model.GetUserThreadsOpts{
+			PageSize:   25,
+			Extended:   false,
+			Deleted:    false,
+			Unread:     true,
+			Since:      0,
+			TotalsOnly: false,
+			Before:     oldestUnreadThreadId,
+		})
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		oldestUnreadThreadId = unreadThreads[len(unreadThreads)-1].PostId
+		// idle time between scrolls, between 1 and 10 seconds.
+		idleTime := time.Duration(1+rand.Intn(10)) * time.Second
+		select {
+		case <-c.stopChan:
+			return control.UserActionResponse{Info: "action canceled"}
+		case <-time.After(idleTime):
+		}
+	}
+
+	return control.UserActionResponse{Info: "Visited Global Threads Screen"}
+}
+
+func (c *SimulController) followThread(u user.User) control.UserActionResponse {
+	collapsedThreads, resp := control.CollapsedThreadsEnabled(u)
+	if resp.Err != nil || !collapsedThreads {
+		return resp
+	}
```

Comment:
Wondering if we can create a middleware function to gate all CRT related calls? That way we don't have to repeat this check everywhere.